src/core/CodeUnit.ml

```diff
@@ -17,9 +17,13 @@ struct
   type t =
     { origin : CodeUnitID.t;
       index : int;
-      name : string option }
+      name : string option;
+      unfolder : t option}
```

Comment:
It seems the definition is more general than it needs to be---it appears to allow:

1. The unfolder being in a different unit.
2. The unfolder having arbitrary names.
3. The unfolder itself having its unfolder (recursively).

Is this overkill? Is it enough to have `unfolder_index : int option`?

Reply:
@favonia I think I need it, but I'm open to ideas. A common pattern that I have is:

1. I have an identifier (like `plus`).
2. Resolve it to a global, `sym : Global.t`.
3. Get the dimension term corresponding to the _unfolder_ of `sym`.

Do you have a suggestion for how to achieve this otherwise?

Reply:
I don't see how your typical usage needs any of the three things I listed. My point is, what would be missing in this code?
```ocaml
let unfolder s =
  match s.unfolder_index with
  | None -> None
  | Some index -> {s with index; unfolder_index = None; name = <something something>}
```
(Or we can take `unfolder_name` as well.)

Reply:
Let me try it...

Reply:
OK, I tried to make this change and this is breaking things in ways I don't understand at the moment... Do you want to have a try?

Reply:
@favonia here's my broken code: https://github.com/RedPRL/cooltt/tree/favonia-unfolder-index

Reply:
Never mind, found the bug! WIll push fix.